selene_sdk/samplers/multi_sampler.py

```diff
@@ -0,0 +1,324 @@
+
+        Raises
+        ------
+        ValueError
+            If `mode` is not a valid mode.
+
+        """
+        if mode not in self.modes:
+            raise ValueError(
+                "Tried to set mode to be '{0}' but the only valid modes are "
+                "{1}".format(mode, self.modes))
+        self.mode = mode
+
+    def _set_batch_size(self, batch_size, mode=None):
+        """
+        Sets the batch size for DataLoader for the specified mode,
+        if the specified batch_size does not equal the current batch_size.
+        Parameters
+        ----------
+        batch_size : int
+            The batch size for the mode.
+        mode : str, optional
+            Default is None. The  mode to set batch_size
+            If None, will use the current mode `self.mode`.
+        """
+        if mode is None:
+            mode = self.mode
+
+        if self._dataloaders[mode]:
+            batch_size_matched = True
+            if self._dataloaders[mode].batch_sampler:
+                if self._dataloaders[mode].batch_sampler.batch_size != batch_size:
+                    self._dataloaders[mode].batch_sampler.batch_size = batch_size
+                    batch_size_matched = False
+            else:
+                if self._dataloaders[mode].batch_size != batch_size:
+                    self._dataloaders[mode].batch_size = batch_size
+                    batch_size_matched = False
+
+            if not batch_size_matched:
+                print("Reset data loader for mode {0} to use the new batch "
+                      "size: {1}.".format(mode, batch_size))
+                self._iterators[mode] = iter(self._dataloaders[mode])
+
+    def get_feature_from_index(self, index):
+        """
+        Returns the feature corresponding to an index in the feature
+        vector.
+
+        Parameters
+        ----------
+        index : int
+            The index of the feature to retrieve the name for.
+
+        Returns
+        -------
+        str
+            The name of the feature occurring at the specified index.
+
+        """
+        return self._index_to_feature[index]
+
+    def sample(self, batch_size=1, mode=None):
+        """
+        Fetches a mini-batch of the data from the sampler.
+
+        Parameters
+        ----------
+        batch_size : int, optional
+            Default is 1. The size of the batch to retrieve.
+        mode : str, optional
+            Default is None. The operating mode that the object should run in.
+            If None, will use the current mode `self.mode`.
+        """
+        mode = mode if mode else self.mode
+        if self._samplers[mode]:
+            return self._samplers[mode].sample(batch_size)
+        else:
+            self._set_batch_size(batch_size, mode=mode)
+            try:
+                data, targets = next(self._iterators[mode])
+                return data.numpy(), targets.numpy()
+            except StopIteration:
```

Comment:
you know how sometimes we set the sampling for RandomPositionsSampler to something extremely large if we wanted training to go on forever until we see plateau? Should we mention that that's an option for users in the RandomPositionsSampler and IntervalsSampler docs when DataLoader is used? Just a thought

Reply:
also similarly `get_validation_set` is missing `Returns` documentation but `get_test_set` has it

Reply:
`DataLoader` now uses a size of `sys.maxsize` (because torch.DataLoader requires it has to have a size for the iterator) so practically it never runs out and even if it runs out we reinitialize it here.  RandomPositionsSampler and IntervalsSampler do not have a size attribute right? I agree it's a good idea to mention it - maybe in TrainModel docs?